Riot/Modules/Room/RoomViewController.m

```diff
@@ -1488,6 +1514,139 @@ - (void)destroy
     [super destroy];
 }
 
+#pragma mark - Start DM
+
+/**
+ Check whether the current room is a direct chat left by the other member.
+ */
+- (void)isDirectChatLeftByTheOther:(void (^)(BOOL isEmptyDirect))onComplete
+{
+    // In the case of a direct chat, we check if the other member has left the room.
+    if (self.roomDataSource)
+    {
+        NSString *directUserId = self.roomDataSource.room.directUserId;
+        if (directUserId)
+        {
+            [self.roomDataSource.room members:^(MXRoomMembers *roomMembers) {
+                MXRoomMember *directUserMember = [roomMembers memberWithUserId:directUserId];
+                if (directUserMember)
+                {
+                    MXMembership directUserMembership = directUserMember.membership;
+                    if (directUserMembership != MXMembershipJoin && directUserMembership != MXMembershipInvite)
+                    {
+                        onComplete(YES);
+                    }
+                    else
+                    {
+                        onComplete(NO);
+                    }
+                }
+                else
+                {
+                    MXLogDebug(@"[RoomViewController] isEmptyDirectChat: the direct user has disappeared");
+                    onComplete(YES);
+                }
+            } failure:^(NSError *error) {
+                MXLogDebug(@"[RoomViewController] isEmptyDirectChat: cannot get all room members");
+                onComplete(NO);
```

Comment:
@ismailgulek you're right
@phlniji I changed my mind since our last discussion about this method `isDirectChatLeftByTheOther`, the completion block should be able to return an error with the boolean, like you suggested. By this way we will be able to alert the user that something wrong happens and stop the process in `restoreDiscussionIfNeeded` method

Reply:
I will change this.